streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -936,7 +950,7 @@ private void resetInvalidOffsets(final InvalidOffsetException e) {
 
                 if (originalReset.equals("earliest")) {
                     addToResetList(partition, seekToBeginning, "No custom setting defined for topic '{}' using original config '{}' for offset reset", "earliest", loggedTopics);
-                } else if (originalReset.equals("latest")) {
```

Comment:
Should we keep this check and add final `else` that throws an `IllegalStateException` instead?

Reply:
Yeah that sounds better :)